run/events-pubsub/Startup.cs

```diff
@@ -46,38 +45,47 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<
         {
             endpoints.MapPost("/", async context =>
             {
-                logger.LogInformation("Handling HTTP POST");
-
-                using (var reader = new StreamReader(context.Request.Body))
+                var cloudEvent = await context.Request.ReadCloudEventAsync();
+                if (cloudEvent == null)
                 {
-                    var body = await reader.ReadToEndAsync();
-                    logger.LogInformation($"HTTP POST body: {body}");
+                    context.Response.StatusCode = 400;
+                    await context.Response.WriteAsync("Bad request: No CloudEvent received");
+                    return;
+                }
+                logger.LogInformation("Received CloudEvent\n" + GetEventLog(cloudEvent));
 
-                    dynamic cloudEventData = JsonConvert.DeserializeObject(body);
-                    if (cloudEventData == null)
-                    {
-                        context.Response.StatusCode = 400;
-                        await context.Response.WriteAsync("Bad request: No Pub/Sub message received");
-                        return;
-                    }
+                var messagePublishedData = CloudEventConverters.ConvertCloudEventData<MessagePublishedData>(cloudEvent);
+                var pubSubMessage = messagePublishedData?.Message;
+                if (messagePublishedData == null || pubSubMessage == null)
+                {
+                    context.Response.StatusCode = 400;
+                    await context.Response.WriteAsync("Bad request: Invalid Pub/Sub message format");
+                    return;
+                }
 
-                    dynamic pubSubMessage = cloudEventData["message"];
-                    if (pubSubMessage == null)
-                    {
-                        context.Response.StatusCode = 400;
-                        await context.Response.WriteAsync("Bad request: Invalid Pub/Sub message format");
-                        return;
-                    }
+                var data = pubSubMessage.Data;
+                logger.LogInformation($"Data: {data.ToBase64()}");

```

Comment:
nit: Why do you show this now?

Reply:
I found it helpful to see the base64 encoded string first, before decoding it later..I don't feel strongly about showing it though and can remove if you want.

Reply:
No, not my heart set on it, if you think it's going to be helpful, that's fine.

Reply:
Ok, I'll keep it then.